lib/matplotlib/backends/backend_agg.py

```diff
@@ -148,18 +148,66 @@ def draw_path(self, gc, path, transform, rgbFace=None):
                     c = c[ii0:ii1]
                     c[0] = Path.MOVETO  # move to end of last chunk
                 p = Path(v, c)
+                p.simplify_threshold = path.simplify_threshold
                 try:
                     self._renderer.draw_path(gc, p, transform, rgbFace)
                 except OverflowError as err:
-                    raise OverflowError(
-                        "Exceeded cell block limit (set 'agg.path.chunksize' "
-                        "rcparam)") from err
+                    msg = ("Exceeded cell block limit in Agg.\n\n"
+                           "Please reduce "
+                           "the value of rcParams['agg.path.chunksize'] "
+                           f"(currently {nmax}) or increase the "
+                           "path simplification threshold"
+                           "(rcParams['path.simplify_threshold'] = "
+                           f"{mpl.rcParams['path.simplify_threshold']:.2f} "
+                           "by default and path.simplify_threshold "
+                           f"= {path.simplify_threshold:.2f} "
+                           "on the input)."
+                           )
+                    raise OverflowError(msg) from err
```

Comment:
I would suppress exception chaining here (`... from None`): the exception is something that we created ourselves anyways (inserted into the C++ agg code in https://github.com/matplotlib/matplotlib/pull/3777), so the fact that there's two exceptions solely arises from the fact that it's too annoying to format the lengthy error message in C++, but from the PoV of the user that's really an implementation detail.